Update egg-insurgency--sandstorm.json

Prophet731 opened the pull request:
# Description

Replace the old image to one that's supported. This solves the problem of "GLIBCXX_3.4.26" missing.

ghcr.io/parkervcp/steamcmd:ubuntu

## Checklist for all submissions

<!-- insert X into the brackets to mark it as done (i.e. [x]). You can click preview to make the links appear clickable. -->

* [X] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [X] Have you tested and reviewed your changes with confidence that everything works?
* [X] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

<!-- If this is an egg update fill these out -->

* [X] You verify that the start command applied does not use a shell script
  * [ ] If some script is needed then it is part of a current yolk or a PR to add one
* [ ] The egg was exported from the panel

QuintenQVD0 commented:
You can not change the .json file directly. Make the change in a panel and then export and fully replace the file fully.

QuintenQVD0 commented:
The feature went a away. ANd you can not edit variables as the startup will not update of already made servers

Prophet731 commented:
> The feature went a away. ANd you can not edit variables as the startup will not update of already made servers

What feature are you talking about?

Also this is what I had to do to allow sandstorm to allow things to work with game stats and such with the command lines.

Prophet731 commented:
Added a fix for the hostname, switched to single quotes instead of double quotes. The shell was interpreting the `!` in the string causing problems.

QuintenQVD0 commented:
I don't think you understand. You updated a variable, and the startup. But if someone updates this egg the variable names do update but the startup does not. So some things then will be empty.

Prophet731 commented:
Fixed my mistake earlier now.

> I don't think you understand. You updated a variable, and the startup. But if someone updates this egg the variable names do update but the startup does not. So some things then will be empty.

The only thing that was removed was the "AUTO_UPDATE" but that isn't used in the startup config. The new variables I added are allowed to be empty so it shouldn't break any older versions.

QuintenQVD0 commented:
> Fixed my mistake earlier now.
> 
> > I don't think you understand. You updated a variable, and the startup. But if someone updates this egg the variable names do update but the startup does not. So some things then will be empty.
> 
> The only thing that was removed was the "AUTO_UPDATE" but that isn't used in the startup config. The new variables I added are allowed to be empty so it shouldn't break any older versions.

Auto update also can not be changed. It is handled by the image internaly not in the startup.

parkervcp commented on game_eggs/steamcmd_servers/insurgency_sandstorm/egg-insurgency--sandstorm.json:
```diff
@@ -24,8 +22,8 @@
-            "script": "apt -y update\r\napt -y --no-install-recommends install curl lib32gcc-s1 ca-certificates\r\n\r\n## just in case someone removed the defaults.\r\nif [[ \"${STEAM_USER}\" == \"\" ]] || [[ \"${STEAM_PASS}\" == \"\" ]]; then\r\n    echo -e \"steam user is not set.\\n\"\r\n    echo -e \"Using anonymous user.\\n\"\r\n    STEAM_USER=anonymous\r\n    STEAM_PASS=\"\"\r\n    STEAM_AU ...
-            "container": "ghcr.io\/parkervcp\/installers:debian",
+            "script": "#!\/bin\/bash\r\n## install required packages to install insurgency: sandstorm\r\napt update\r\napt -y --no-install-recommends install curl unzip libstdc++6 lib32gcc1 ca-certificates\r\n## install steamcmd\r\ncd \/tmp\r\ncurl -sSL -o steamcmd.tar.gz http:\/\/media.steampowered.com\/installer\/steamcmd_linux.tar.gz\r\nmkdir -p \/mnt\/server\/steam\r\ntar -xzvf steamcmd.tar.g ...
+            "container": "ubuntu:20.04",
```
why even change this?

gOOvER replied:
becaus Debian 11 have only version 2.31-13+deb11u7 of glibc

QuintenQVD0 replied:
In the install container? Wy is that there even needed. Notthing runs there.

gOOvER replied:
ok, i didnt follow the whole conversation. He changed ste steamcmd image to ubuntu. This should be enough, because its on 22.04

So the install image dont need a change.

But @Quinten; it would be better to update all images to Debian 12. i take a look into your PR, and as far i see, its ok. Only some packages need changes

Prophet731 replied:
> why even change this?

It's been a while since I even touched the install script side so I don't remember why that would be different.

gOOvER replied:
> > why even change this?
> 
> It's been a while since I even touched the install script side so I don't remember why that would be different.

you can stay on the installer image. The game need glibc only, when the server is running and you allread chnge the steamcmd image to ubuntu, which is on 22.04

QuintenQVD0 replied:
> ok, i didnt follow the whole conversation. He changed ste steamcmd image to ubuntu. This should be enough, because its on 22.04
> 
> So the install image dont need a change.
> 
> But @Quinten; it would be better to update all images to Debian 12. i take a look into your PR, and as far i see, its ok. Only some packages need changes

You have me on discord. Some eggs will need updates to the install image like nodejs. Did not have time yet to make it.

QuintenQVD0 reviewed (approved):
LGTM! Thank you.
